lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -826,10 +825,7 @@ private function computeAssociationChanges($assoc, $value)
         if ($value instanceof PersistentCollection && $value->isDirty()) {
             $coid = spl_object_hash($value);
 
-            if ($assoc['isOwningSide']) {
-                $this->collectionUpdates[$coid] = $value;
-            }
-
+            $this->collectionUpdates[$coid] = $value;
```

Comment:
Yes, but you're indeed updating a collection adding/removing new items, so it should be checked depending of CollectionPersister implementation.
In most cases, it'll just return, except on CacheCollectionPersister, which would evict the collection (addressing the bug)